org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProviderTest.java

```diff
@@ -184,6 +184,30 @@ public void fixAddActionForGrammarWithAlias() throws Exception {
 				"Add actions to ensure object creation");
 	}
 
+	@Test
+	public void testConvertTerminalFragmentToTerminalRule() throws Exception {
+		assertAndApplySingleResolution(editorForGrammar("Model: digit=ABC;", "terminal fragment ABC: '0'..'9';"),
+				INVALID_TERMINAL_FRAGMENT_RULE_REFERENCE, 0, "Convert terminal fragment to terminal rule");
+	}
+
+	@Test
+	public void testFixInvalidTerminalFragmentHiddenToken() throws Exception {
+		assertAndApplySingleResolution(editorForGrammar("Model hidden(ABC): a=ID;", "terminal fragment ABC: '0'..'9';"),
```

Comment:
Do we also get the `Convert terminal fragment to terminal rule` quickfix for this case?

Reply:
I have enhanced the code to show the above quick-fix.